mozilla_django_oidc/auth.py

```diff
@@ -25,6 +44,32 @@ def __init__(self, *args, **kwargs):
 
         self.UserModel = get_user_model()
 
+    def create_user(self, email, **kwargs):
+        """Return object for a newly created user account."""
+
+        # bluntly stolen from django-browserid
+        # https://github.com/mozilla/django-browserid/blob/master/django_browserid/auth.py
+        from django.db import IntegrityError  # Importing at the top causes issues on DB init.
+
+        username_algo = import_from_settings('OIDC_USERNAME_ALGO', False)
+
+        if username_algo:
+            username = username_algo(email)
+        else:
+            username = default_username_algo(email)
+
+        try:
+            return self.UserModel.objects.create_user(username, email)
+        except IntegrityError as err:
+            # Race condition! Attempt to re-fetch from the database.
```

Comment:
This is probably more of a ``browserid`` side effect. I suggest we omit this try/except until we have actual cases where django throws an ``IntegrityError``.